src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -2735,42 +2761,68 @@ export default class AircraftModel {
         delete this.conflicts[conflictingAircraft.callsign];
     }
 
-    // TODO: needs better name
     /**
      * @for AircraftModel
-     * @method _contactAircraftAfterControllabilityChange
+     * @method _updateAircraftControllability
+     * @param isAutoTower {boolean}
      * @private
      */
-    _contactAircraftAfterControllabilityChange() {
-        // Crossing into the center
+    _updateAircraftControllability(isAutoTower) {
+        if (this.projected) {
+            return;
+        }
+
+        const newControllability = (isAutoTower && this._isUnderTowerControl()) ?
```

Comment:
I find this name confusing. Taking a stab at it, does this mean an aircraft that was not controllable, now is (and visa versa)?

Could `isNewlyControllable` work here instead? Or am I completely misunderstanding what this is doing?

Reply:
We want to detect a change from controllable -> not controllable or vice versa.

`this.isControllable` is the "old" cached value from before. `newControllability` is our freshly calculated value as at the current simulation tick. This variable used to be called `isInsideAirspace` (see [previous version of the whole function](https://github.com/openscope/openscope/blob/94b8cee2bca8470186adf4d5a5ec25438c6e9c10/src/assets/scripts/client/aircraft/AircraftModel.js#L2756-L2774)) but that name is no longer accurate as the test has become a bit more involved. A couple of lines down we compare the two and if needed, update `this.isControllable` to the new value and react to the change by  doing other stuff.

`isNewlyControllable` seems to mean something different, namely
```
// was not controllable before; has just become controllable
const isNewlyControllable = !this.isControllable && newControllability;

// compare with
const isNewlyUncontrollable = this.isControllable && !newControllability;
```

Reply:
I see what you're saying and that helps me understand this much better. Yeah, I thought `newControllability` was a flag of some sort, but instead it's the next value for `isControllable`. I think the word `new` through me off here. Would the name `nextControllability` work here? That name helps suggest that this value is going to be used to replace something later on.

Reply:
`nextControllability` makes sense, done